File: features/main/src/main/java/com/inquallity/daggersandbox/main/MainPresenter.java
package com.inquallity.daggersandbox.main;

import android.content.Context;

import androidx.annotation.NonNull;

import com.inquallity.daggersandbox.core.crossmodule.CrossModule;
import com.inquallity.daggersandbox.core.domain.MainInteractor;

import javax.inject.Inject;

public class MainPresenter {

    @NonNull
    private final MainInteractor interactor;
    @NonNull
    private final CrossModule crossModule;

    @Inject
    public MainPresenter(@NonNull final MainInteractor interactor, @NonNull final CrossModule crossModule) {
        this.interactor = interactor;
        this.crossModule = crossModule;
    }

    public void initialize() {
        interactor.init();
    }

    public void doSmth() {
        interactor.doSomething();
    }

    public void moveToSecondary(Context ctx) {
        crossModule.toSecond(ctx);
    }

}

File: features/main/src/main/java/com/inquallity/daggersandbox/main/MainPresenter.kt
package com.inquallity.daggersandbox.main

import android.content.Context
import com.inquallity.daggersandbox.core.crossmodule.CrossModule
import com.inquallity.daggersandbox.core.domain.MainInteractor
import javax.inject.Inject

class MainPresenter @Inject constructor(
    private val interactor: MainInteractor,
    private val crossModule: CrossModule
) {

    fun initialize() {
        interactor.init()
    }

    fun doSmth() {
        interactor.doSomething()
    }

    fun moveToSecondary(ctx: Context) {
        crossModule.toSecond(ctx)
    }
}